File: lib/core/model/place.dart
import 'package:isar/isar.dart';

part 'place.g.dart';

@collection
class Place {
  Place({
    required this.latitude,
    required this.longitude,
    required this.emoji,
    required this.note,
    required this.createdAt,
  });

  Id id = Isar.autoIncrement;
  double latitude;
  double longitude;
  String emoji;
  String note;
  DateTime createdAt;
}

File: web/src/core/model/Place.ts
export interface Place {
  id: number;
  latitude: number;
  longitude: number;
  emoji: string;
  note: string;
  folderName?: string;
  createdAt: Date;
}

export function createPlace(
  latitude: number,
  longitude: number,
  emoji: string,
  note: string,
  folderName?: string
): Omit<Place, 'id' | 'createdAt'> {
  return {
    latitude,
    longitude,
    emoji,
    note,
    folderName,
  };
}